Duplicati/Library/Utility/Utility.cs

```diff
@@ -35,6 +35,14 @@ public static class Utility
         /// </summary>
         public static long DEFAULT_BUFFER_SIZE => SystemContextSettings.Buffersize;
 
+        /// <summary>
+        /// Boolean value parsed from the environment variable FILESYSTEM_CASE_SENSITIVE,
+        /// or null if it is not a valid boolean string.
+        /// </summary>
+        public static bool? EnvVar_FileSystemCaseSensitive = Utility.ParseBool(
+            Environment.GetEnvironmentVariable("FILESYSTEM_CASE_SENSITIVE"), null

```

Comment:
This appears to have triggered many failures in Travis and AppVeyor...

If the `FILESYSTEM_CASE_SENSITIVE` variable is not set, `GetEnvironmentVariable("FILESYSTEM_CASE_SENSITIVE")` will return `null`.  In this case, the call to `Utility.ParseBool` will then return the default value `null`.  This is semantically different than before, where the default value provided to `Utility.ParseBool` is determined by `Utility.IsClientLinux && !Utility.IsClientOSX`.

Reply:
Thanks for looking into this. It triggered failures because the code wasn't correct (I was misusing `Utility.ParseBool()`). However, I don't quite get why my change would change the semantics.

Now I switched to a different way of implementation: Fully caching the whole thing. I hope it will work :)